weblab_cg/_generator_chaste_cg.py

```diff
@@ -0,0 +1,627 @@
+                # Convert voltage to mV, time to mS and calcium to millimolar
+                dividend_unit = self._get_desired_units(d_eqs[i].lhs.args[0])
+                divisor_unit = self._get_desired_units(d_eqs[i].lhs.args[1][0])
+                units = dividend_unit / divisor_unit
+                units_from = self._model.units.summarise_units(d_eqs[i].lhs)
+                factor = self._model.units.get_conversion_factor(units, from_unit=units_from)
+            if d_eqs[i].lhs == self._membrane_stimulus_current:
+                factor = 1.0
+                # strip out var from time variable name as printing will add it back later
+                area = sp.sympify(stimulus_sign +
+                                  'GetIntracellularAreaStimulus(' +
+                                  self._var_chaste_interface_printer.doprint(self._time_variable)[3:] + ')')
+
+                if self._use_capacitance_i_ionic:
+                    rhs_divider = '/ HeartConfig::Instance()->GetCapacitance()'
+                if self._use_capacitance_i_ionic and self._membrane_capacitance is not None:
+                    stim_current_eq_rhs = area * self._membrane_capacitance.lhs
+                    if self._membrane_capacitance_factor != 1.0:
+                        stim_current_eq_rhs *= 1 / self._membrane_capacitance_factor
+                else:
+                    stim_current_eq_rhs = area
+                    if self._ionic_current_factor != 1.0:
+                        stim_current_eq_rhs *= 1 / self._ionic_current_factor
+
+                d_eqs[i] = sp.Eq(d_eqs[i].lhs, sp.simplify(stim_current_eq_rhs))
+
+            # Format the equation and put it at the front of the result list to keep order (we're looping backwards)
+            equations.insert(0, {'lhs': d_eqs[i].lhs,
+                                 'factor': factor,
+                                 'rhs': d_eqs[i].rhs,
+                                 'units': units,
+                                 'rhs_divider': rhs_divider,
+                                 'in_membrane_voltage': d_eqs[i] not in self._derivative_eqs_exlc_voltage,
+                                 'is_voltage': is_voltage})
+
+        for i in range(len(equations)):
+            if equations[i]['factor'] != 1.0:
+                for j in range(i + 1, len(equations)):
+                    equations[j]['rhs'] = equations[j]['rhs'].subs({equations[i]['lhs']:
+                                                                    (1 / equations[i]['factor'] * equations[i]['lhs'])})
+            if float(equations[i]['factor']) != 1.0:
+                equations[i]['rhs'] = sp.simplify(equations[i]['factor'] * equations[i]['rhs'])
+            equations[i]['rhs'] = \
+                self._printer.doprint(equations[i]['rhs']) + equations[i]['rhs_divider']
+            equations[i]['lhs'] = self._printer.doprint(equations[i]['lhs'])
+
+        return equations
+
+    def _format_system_info(self):
```

Comment:
- Maybe `ode_system` instead of `system`? I was expecting it to show python versions, operating system etc.
- The method doesn't format info (`format_info(self, info)`) but generates and returns info, in some complex type. Would update the name and docstring to describe what this thing returns.
- The code below isn't very easy to read so please add some comments explaining what it's supposed to do / what the expected output is

Reply:
(as in, imagine something happens upstream so that this method stops working, and I want to update it, then I need to know what the expected output is)

Reply:
done